lib/hammer_cli_foreman/smart_class_parameter.rb

```diff
@@ -97,8 +111,53 @@ class UpdateCommand < HammerCLIForeman::UpdateCommand
             ['string', 'boolean', 'integer', 'real', 'array', 'hash', 'yaml', 'json'])
       option "--validator-type", "VALIDATOR_TYPE", _("Type of the validator."),
         :format => HammerCLI::Options::Normalizers::Enum.new(['regexp', 'list', ''])
+
+      validate_options do
+        if option(:option_name).exist?
+          any(:option_puppetclass_name, :option_puppetclass_id).required
+        end
+      end
+    end
+
+    class AddOverrideValueCommand < HammerCLIForeman::CreateCommand
+      resource :override_values
+      command_name 'add-override-value'
+
+      success_message _("Override value created")
+      failure_message _("Could not create the override_value")
+
+      build_options do |o|
```

Comment:
nitpick - meaningless variable names are frowned upon in Ruby standards, should be called option instead of o